app/src/editor/shared/components/Selection.jsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * Manages current selection.
+ * Selected item is an id.
+ * Can add/remove from selection, select all or select none.
+ */
+
+import React from 'react'
+
+/* eslint-disable react/no-unused-state */
+const SelectionContext = React.createContext({})
+const { Consumer } = SelectionContext
+
+export { Consumer, SelectionContext }
+export default Consumer
+
+export class SelectionProvider extends React.Component {
+    componentDidMount() {
+        window.addEventListener('keydown', this.updateKeys)
+        window.addEventListener('keyup', this.updateKeys)
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('keydown', this.updateKeys)
+        window.removeEventListener('keyup', this.updateKeys)
+    }
+
+    updateKeys = (event) => {
+        this.shiftKey = event.shiftKey
+        this.ctrlKey = event.ctrlKey
```

Comment:
Do we have to keep track on both? How about:
```js
this.keep = event.shiftKey || event.ctrlKey
```

Reply:
Intention is to keep the business logic about what should happen in the `onEvent` handler. This handler just collects state but makes no decisions.

Reply:
Cool.